Yafc.Model/Model/Project.cs

```diff
@@ -148,6 +148,29 @@ public class Project : ModelObject {
             page.MarkAsDeleted();
             _ = this.RecordUndo().pages.Remove(page);
         }
+
+        /// <summary>
+        /// Get the page that is visually next (i.e. to the right of the current selected page on the tab bar)
+        /// from the specified one.
+        /// </summary>
+        /// <param name="currentPage">The page to get the next page from (probably the current page).</param>
+        /// <returns>The page object that should be set to active.</returns>
+        public ProjectPage? VisibleNeighborOfPage(ProjectPage? currentPage, bool forward = true) {
+            if (currentPage == null) {
+                return null;
```

Comment:
No idea if there are situations where `currentPage` is not defined. But we could return the 'first' page in these cases? (assuming there is at least one page ofc)

Reply:
We could. I'd be a little concerned that that would only happen in an out-of-the-way interaction pattern somewhere I wasn't familiar enough with the system to test; I think I'd rather keep it returning null in that case, which hopefully would mean we just don't change page at all.

Reply:
Hmm, do I understand it correctly that the `null` case is the unexpected behavior? If so, would we benefit from logging it, or is it not even remotely possible to get anything useful out of loggin this event? Just wondering.

Reply:
`null` would just close the active page (in [`MainScreen.ChangePage()`](https://github.com/shpaass/yafc-ce/blob/5a8d91274716828543902dc5663ab67fa682b9c7/Yafc/Windows/MainScreen.cs#L152)) showing compilatron on the background.

Reply:
What I meant is the following:
If this function is getting `null` as input, is it normal behavior, or is it unintended behavior to the point where it should not have happened and we should report it?

Reply:
`currentPage` is `null` when there is no page selected, could be when there are no pages, or the user (somehow) does not have a page selected (just started YAfc for example).

That is why I though, we might as well open the 'first' (whatever that means :wink: ) page (assuming there is one)

Reply:
Leaving as is, is also fine by me. Maybe @shpaass has some opinion on the best behavior for such scenarios?

Reply:
Hmm, don't like that. Plus it also means that if you're currently on compilatron, and do have pages you could switch to, you can't switch to them.

What about:
- If the current page is null (you're viewing the compilatron, e.g. after deleting a page) we'll go to the first page as you suggested
- At that point, if `VisibleNeighborOfPage` returns null, it means you have no pages, so switching to compilatron is presumably fine

Reply:
Sounds good

Reply:
Gave that an implementation pass. Note the additional change to `RemovePage`, which I think fixes a latent bug - we weren't removing that page from the display pages list. It wouldn't get rendered in the tab bar because it _would_ get removed from the Guid lookup table, but when iterating through `displayPages` directly we'd catch it.

I did test undo, and you are still able to do it correctly. Sad about not being able to preserve the method chaining though.